api/proto/banyandb/stream/v1/query.proto

```diff
@@ -54,7 +55,7 @@ message QueryResponse {
 // QueryRequest is the request contract for query.
 message QueryRequest {
   // metadata is required
-  common.v1.Metadata metadata = 1;
+  common.v1.Metadata metadata = 1 [(validate.rules).message.required = true];
```

Comment:
`projection` is required and contains at least one item, otherwise nothing will be returned